runtime/vm/LayoutFFITypeHelpers.cpp

```diff
@@ -100,31 +101,93 @@ LayoutFFITypeHelpers::getArrayFFIType(char **layout, UDATA nElements) {
 	return typeFFI;
 }
 
-ffi_type*
+ffi_type *
 LayoutFFITypeHelpers::getStructFFIType(char **layout)
 {
+	char *currentLayout = *layout;
+	J9LayoutStrFFITypeEntry *layoutStrFFITypeEntry = NULL;
+	J9LayoutStrFFITypeEntry *resultEntry = NULL;
+	ffi_type **structElements = NULL;
 	ffi_type *typeFFI = NULL;
+	PORT_ACCESS_FROM_JAVAVM(_vm);
+
+	if (NULL == _vm->layoutStrFFITypeTable) {
+		_vm->layoutStrFFITypeTable = createLayoutStrFFITypeTable(_vm);
+		if (NULL == _vm->layoutStrFFITypeTable) {
+			goto freeAllMemoryThenExit;
+		}
+	}
+
+	layoutStrFFITypeEntry = (J9LayoutStrFFITypeEntry *)j9mem_allocate_memory(sizeof(J9LayoutStrFFITypeEntry), J9MEM_CATEGORY_VM_FFI);
+	if (NULL == layoutStrFFITypeEntry) {
+		goto freeAllMemoryThenExit;
+	}
+
+	layoutStrFFITypeEntry->layoutStrLength = getLengthOfStructLayout(currentLayout);
+	layoutStrFFITypeEntry->layoutStr = (U_8 *)j9mem_allocate_memory(layoutStrFFITypeEntry->layoutStrLength + 1, J9MEM_CATEGORY_VM_FFI);
```

Comment:
Malformed layout strings should be caught here:
```c
	layoutStrFFITypeEntry->layoutStrLength = getLengthOfStructLayout(currentLayout);
	if (0 == layoutStrFFITypeEntry->layoutStrLength) {
		/* malformed input */
		goto freeAllMemoryThenExit;
	}
```

Reply:
Fixed given the code in `getLengthOfStructLayout` is updated as above.